plugins/permission-react/src/components/PermissionedRoute.tsx

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { ComponentProps, ReactElement } from 'react';
+import { Route } from 'react-router';
+import { useApp } from '@backstage/core-plugin-api';
+import { usePermission } from '../hooks';
+import { Permission } from '@backstage/plugin-permission-common';
+
+/**
+ * Returns a React Router Route which only renders the element when authorized. If unathorized, the Route will render a
+ * NotFoundErrorPage (see {@link @backstage/core-app-api#AppComponents}).
+ * @public
+ */
+export const PermissionedRoute = ({
```

Comment:
Did we workshop this name to the limit? x)

Thinking about some options like `GuardedRoute`, `ProtectedRoute`, `PolicedRoute`. Don't know if I immediately have any favorites there and `PermissionedRoute` kinda seems like the most clear in the end, just checking x)

Reply:
I'm not 100% sure on the history of this name (I think it used to be `AuthenticatedRoute` very early on before we moved everything to `Permission-`?) but I'll let @mtlewis chime in here. Personally I do agree that `PermissionedRoute` seems the most clear and aligned with the rest of the framework that we provide.

Reply:
+1 - I like the inclusion of the "permission" root, it ties it back to the permission plugins. There hasn't been a ton of thought put into these names to date though so very happy to discuss more!